cobbler/utils/filesystem_helpers.py

```diff
@@ -0,0 +1,347 @@
+    cache: bool = True,
+    api=None,
+):
+    """
+    Copy 1 or more files with a pattern into a destination.
+
+    :param pattern: The pattern for finding the required files.
+    :param dst: The destination for the file(s) found.
+    :param require_match: If the glob pattern does not find files should an error message be thrown or not.
+    :param symlink_ok: If it is okay to just use a symlink to link the file to the destination.
+    :param cache: If it is okay to use a file from the cache (which could be possibly newer) or not.
+    :param api: Passed to ``linkfile()``.
+    :raises CX: Raised in case files not found according to ``pattern``.
+    """
+    files = glob.glob(pattern)
+    if require_match and not len(files) > 0:
+        raise CX("Could not find files matching %s" % pattern)
+    for file in files:
+        dst1 = os.path.join(dst, os.path.basename(file))
+        linkfile(file, dst1, symlink_ok=symlink_ok, cache=cache, api=api)
+
+
+def rmfile(path: str):
+    """
+    Delete a single file.
+
+    :param path: The file to delete.
+    """
+    try:
+        os.remove(path)
+        logger.info('Successfully removed "%s"', path)
+    except FileNotFoundError:
+        pass
+    except OSError as ioe:
+        logger.warning('Could not remove file "%s": %s', path, ioe.strerror)
+
+
+def rmtree_contents(path: str):
+    """
+    Delete the content of a folder with a glob pattern.
+
+    :param path: This parameter presents the glob pattern of what should be deleted.
+    """
+    what_to_delete = glob.glob("%s/*" % path)
+    for x in what_to_delete:
+        rmtree(x)
+
+
+def rmtree(path: str) -> Optional[bool]:
+    """
+    Delete a complete directory or just a single file.
+
+    :param path: The directory or folder to delete.
+    :return: May possibly return true on success or may return None on success.
+    :raises CX: Raised in case ``path`` does not exist.
+    """
+    # TODO: Obsolete bool return value
+    try:
+        if os.path.isfile(path):
+            return rmfile(path)
+        logger.info("removing: %s", path)
+        return shutil.rmtree(path, ignore_errors=True)
+    except OSError as ioe:
+        log_exc()
+        if ioe.errno != errno.ENOENT:  # doesn't exist
+            raise CX("Error deleting %s" % path) from ioe
+        return True
+
+
+def rmglob_files(path: str, glob_pattern: str):
+    """
+    Deletes all files in ``path`` with ``glob_pattern`` with the help of ``rmfile()``.
+
+    :param path: The folder of the files to remove.
+    :param glob_pattern: The glob pattern for the files to remove in ``path``.
+    """
+    for p in Path(path).glob(glob_pattern):
+        rmfile(str(p))
+
+
+def mkdir(path, mode=0o755):
+    """
+    Create directory with a given mode.
+
+    :param path: The path to create the directory at.
+    :param mode: The mode to create the directory with.
+    :raises CX: Raised in case creating the directory fails with error code 17.
```

Comment:
Desciption doesn't look correct. According to the code it raised in case if error is other than 17, so it actually doesn't fail if the directory already exists.